packages/datadog-plugin-fs/src/index.js

```diff
@@ -0,0 +1,490 @@
+}
+
+function createWrapDirClose (config, tracer, sync) {
+  const name = sync ? 'fs.dir.closesync' : 'fs.dir.close'
+  return function wrapDirClose (close) {
+    function options () {
+      const tags = makeFSTags(this.path, null, config, tracer)
+      return { tags }
+    }
+    return tracer.wrap(name, options, close)
+  }
+}
+
+function createWrapDirAsyncIterator (config, tracer, instrumenter) {
+  return function wrapDirAsyncIterator (asyncIterator) {
+    return function wrappedDirAsyncIterator () {
+      if (!kDirReadPromisified) {
+        const keys = Reflect.ownKeys(this)
+        for (const key of keys) {
+          if (kDirReadPromisified && kDirClosePromisified) break
+          if (typeof key !== 'symbol') continue
+          if (!kDirReadPromisified && getSymbolName(key).includes('kDirReadPromisified')) {
+            kDirReadPromisified = key
+          }
+          if (!kDirClosePromisified && getSymbolName(key).includes('kDirClosePromisified')) {
+            kDirClosePromisified = key
+          }
+        }
+      }
+      instrumenter.wrap(this, kDirReadPromisified, createWrapDirRead(config, tracer))
+      instrumenter.wrap(this, kDirClosePromisified, createWrapKDirClose(config, tracer, instrumenter))
+      return asyncIterator.call(this)
+    }
+  }
+}
+
+function createWrapKDirClose (config, tracer, instrumenter) {
+  return function wrapKDirClose (kDirClose) {
+    return function wrappedKDirClose () {
+      const tags = makeFSTags(this.path, null, config, tracer)
+      return tracer.trace('fs.dir.close', { tags }, (span) => {
+        const p = kDirClose.call(this)
+        const unwrapBoth = () => {
+          instrumenter.unwrap(this, kDirReadPromisified)
+          instrumenter.unwrap(this, kDirClosePromisified)
+        }
+        p.then(unwrapBoth, unwrapBoth)
+        return p
+      })
+    }
+  }
+}
+
+function createOpenTags (config, tracer) {
+  return function openTags (path, flag, mode) {
+    if (!flag || typeof flag === 'function') {
+      flag = null
+    }
+    return makeFSFlagTags(path, { flag }, 'r', config, tracer)
+  }
+}
+
+function createCloseTags (config, tracer) {
+  return function closeTags (fd) {
+    if (typeof fd !== 'number' || !Number.isInteger(fd)) {
+      return
+    }
+    return makeFSTags(fd, null, config, tracer)
+  }
+}
+
+function createReadFileTags (config, tracer) {
+  return function readFileTags (path, options) {
+    return makeFSFlagTags(path, options, 'r', config, tracer)
+  }
+}
+
+function createWriteFileTags (config, tracer) {
+  return function writeFileTags (path, data, options) {
+    return makeFSFlagTags(path, options, 'w', config, tracer)
+  }
+}
+
+function createAppendFileTags (config, tracer) {
+  return function appendFileTags (path, data, options) {
+    return makeFSFlagTags(path, options, 'a', config, tracer)
+  }
+}
+
+function createCopyFileTags (config, tracer) {
+  return function copyFileTags (src, dest, flag) {
+    src = src || 'undefined'
```

Comment:
Let's leave it empty which will not add the tag. This will avoid confusion if a file is named `undefined`.